tests/ui/unnecessary_find_map.rs

```diff
@@ -21,3 +21,7 @@ fn main() {
 fn find_map_none_changes_item_type() -> Option<bool> {
     "".chars().find_map(|_| None)
 }
+
+fn issue11260() {
+    let _x = std::iter::once(1).find_map(|n| (n > 1).then_some(n));
```

Comment:
Could you add some more tests? :cowboy_hat_face:

Reply:
Do you have anything specific in mind? I realized that this fix also applies to `unnecessary_filter_map`, so I'm going to add a test that it also does the right thing there, but I'm not sure what else we would want to test here

Reply:
Added a test for unnecessary_filter_map and testing that it doesn't incorrectly suggest `.find()` when calling `.then_some()` with a different option